### PR TITLE
docs: local cluster environment with Argo Workflows

### PR DESCRIPTION
**Pull request by paulfouquet**
#### Motivation

It is useful to be able to deploy a Kubernetes cluster locally to do some tests without having to rely on AWS.

#### Modification

Document how to install a K8s cluster locally using `kind`.

#### Checklist

- [ ] Tests updated NA
- [x] Docs updated
- [ ] Issue linked in Title NA

### CODE REVIEW

File: infra/local.environment.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Local environment
+
+It is possible to install a cluster on your local machine. In this documentation we will use [`kind`](https://kind.sigs.k8s.io/).
+
+## Pre-requisite
+
+- [Docker](https://docs.docker.com/engine/install/) (preferred as production environment uses it) or Podman
```
Comment (blacha) [inline infra/local.environment.md:7]: EKS doesn't use docker, I believe its all `containerd` now? 
but using docker is preferred

File: infra/local.environment.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+> **_NOTE:_** The [`kind` Node container](https://hub.docker.com/r/kindest/node/tags) version number matches the Kubernetes version (`kindest/node:vX.Y.Z` uses Kubernetes `vX.Y.Z`). To specify a Kubernetes version for your cluster, use `--image kindest/node:vX.Y.Z`. It is a good idea to use the same version as the production environment (`version` in [`LinzEksCluster` Stack](../infra/eks/cluster.ts)).
+
+## Argo Workflows
+
+Choose and install the Controller and Server of [a release](https://github.com/argoproj/argo-workflows/releases/) that matches the [production version (`appVersion`)](https://github.com/linz/topo-workflows/blob/master/infra/charts/argo.workflows.ts). The installation process should be described in the Argo Workflow Release page, for example for `v3.5.5`:
+
+```bash
+kubectl create namespace argo
+kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.5.5/install.yaml
```
Comment (blacha) [inline infra/local.environment.md:28]: What would need to change to allow our cdk8s to be deployed into the kind cluster?

File: infra/local.environment.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+You should now be able to run a workflow, for example:
+
+```bash
+argo -n argo submit workflows/test/env.yaml
```
Comment (l0b0) [inline infra/local.environment.md:34]: Do we need to set up a runner of some sort as well? I've tried the following:

```bash
nix-shell --packages argo kind kubectl
kind create cluster --name argo
kubectl create namespace argo
kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.5.5/install.yaml
argo -n argo submit workflows/test/env.yaml
```

It creates a workflow, but it stays in the "Running" state seemingly forever:

```console
❯ argo list --all-namespaces 
NAMESPACE   NAME             STATUS    AGE   DURATION   PRIORITY   MESSAGE
argo        test-env-fbmp7   Running   4m    4m         0          
argo        test-env-48nwz   Running   5m    5m         0          
```

File: infra/local.environment.md (hunk 2 above)
Comment (paulfouquet) [reply infra/local.environment.md:28]: The current deployment using `cdk8s` is using data from our AWS accounts. We can work on making that optional in a separate PR.

File: infra/local.environment.md (hunk 3 above)
Comment (paulfouquet) [reply infra/local.environment.md:34]: My bad, you'll need to replace the image source to a github package one (something not on our AWS ECR, as the local cluster won't have access to it)

File: infra/local.environment.md (hunk 2 above)
Comment (paulfouquet) [reply infra/local.environment.md]: I made the doc Argo Workflows related

File: docs/local.environment.md [hunk 4]
```diff
@@ -31,9 +31,9 @@ kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/dow
 You should now be able to run a workflow, for example:
 
 ```bash
-argo -n argo submit workflows/test/env.yaml
+argo -n argo submit https://raw.githubusercontent.com/argoproj/argo-workflows/refs/heads/main/examples/hello-world.yaml
```
Comment (l0b0) [inline docs/local.environment.md:34]: Adding `--wait` could be useful to show that it finishes successfully. I'm having some trouble with that:

```
❯ argo -n argo submit --wait https://raw.githubusercontent.com/argoproj/argo-workflows/refs/heads/main/examples/hello-world.yaml
Name:                hello-world-475ql
Namespace:           argo
ServiceAccount:      unset (will run with the default ServiceAccount)
Status:              Pending
Created:             Thu Oct 24 08:41:55 +1300 (now)
Progress:            
hello-world-475ql Error at 2024-10-24 08:42:05 +1300 NZDT
❯ argo -n argo get hello-world-475ql
Name:                hello-world-475ql
Namespace:           argo
ServiceAccount:      unset (will run with the default ServiceAccount)
Status:              Error
Message:             Error (exit code 1): pods "hello-world-475ql" is forbidden: User "system:serviceaccount:argo:default" cannot patch resource "pods" in API group "" in the namespace "argo"
Conditions:          
 PodRunning          False
 Completed           True
Created:             Thu Oct 24 08:41:55 +1300 (1 minute ago)
Started:             Thu Oct 24 08:41:55 +1300 (1 minute ago)
Finished:            Thu Oct 24 08:42:05 +1300 (1 minute ago)
Duration:            10 seconds
Progress:            0/1
ResourcesDuration:   0s*(1 cpu),5s*(100Mi memory)

STEP                  TEMPLATE     PODNAME            DURATION  MESSAGE
 ⚠ hello-world-475ql  hello-world  hello-world-475ql  4s        Error (exit code 1): pods "hello-world-475ql" is forbidden: User "system:serviceaccount:argo:default" cannot patch resource "pods" in API group "" in the namespace "argo"  
```

**Review by l0b0 — COMMENTED** [review]
In addition to the problem noted in the comment, I'm also running into an issue where I'm not authorised to work with the existing cluster after rebooting:

```console
❯ kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.5.5/install.yaml
error: error validating "https://github.com/argoproj/argo-workflows/releases/download/v3.5.5/install.yaml": error validating data: failed to download openapi: the server has asked for the client to provide credentials; if you choose to ignore these errors, turn validation off with --validate=false
```

File: docs/local.environment.md (hunk 4 above)
Comment (paulfouquet) [reply docs/local.environment.md:35]: There was a missing step in the cluster configuration. I've added that.

**Review by l0b0 — APPROVED** [review]

**Pull request merged**